### PR TITLE
Nrf security cleanups

### PR DESCRIPTION
**Pull request by SebastianBoe**
Misc. cleanups done to understand nrf_security build scripts.

See commit messages for details.

### CODE REVIEW

File: nrf_security/cmake/kconfig_psa_configure.cmake
Comment (SebastianBoe) [inline nrf_security/cmake/kconfig_psa_configure.cmake:15]: Oops, should have deleted this file.

File: nrf_security/CMakeLists.txt
Comment (frkv) [inline nrf_security/CMakeLists.txt:27]: How can this be set to false in a TF-M build?

File: nrf_security/CMakeLists.txt
Comment (frkv) [inline nrf_security/CMakeLists.txt:35]: I don't know all the details on the reasoning behind the nrf_security_target, but I am curious how (and if) this can be removed.

Also curious how this will run in the dual execution of the nrf_security (once in NSPE with zephyr build system, once inside TFM/SPE build without zephyr build system)

Comment (tejlmand) [reply nrf_security/CMakeLists.txt:35]: I assume this can be cleaned up as @SebastianBoe proposes 👍 

A bit of history.
The reason for this was because originally we could not pass TF-M CMake options directly to TF-M build, but only through Kconfig.
Notice this code is from March 11 (PR: #426, March 9):
https://github.com/nrfconnect/sdk-nrfxlib/commit/ab66d4ac69c14158d5b143a9a1a49ef84aa75b23

and the TFM_CMAKE_OPTIONS came in place in May 7:
https://github.com/zephyrproject-rtos/zephyr/pull/34868

In the Kconfig indirection initial code, it was passed like this:
https://github.com/nrfconnect/sdk-nrfxlib/blob/ab66d4ac69c14158d5b143a9a1a49ef84aa75b23/nrf_security/Kconfig#L1665-L1667

thus the reason for `nrf_security_target`.

So I think this can and should be cleaned up.

**Review by tejlmand — CHANGES_REQUESTED** [review]
Thanks for putting this together.

Careful on the docs though 😉

File: nrf_security/CMakeLists.txt
Comment (tejlmand) [reply nrf_security/CMakeLists.txt:35]: maybe part of the decription I just provided above would be useful in the commit message as to why this code was looking like this, and why it can now bbe cleaned up.

Can help the next one going through commit history / blame to understand the history behind this cleanup.

Comment (frkv) [reply nrf_security/CMakeLists.txt:35]: Excellent to be able to remove complex generator expressions and additional targets!

Looking forward to trying to clean the code up more and make it more readable!

File: nrf_security/CMakeLists.txt
Comment (SebastianBoe) [reply nrf_security/CMakeLists.txt:27]: I don't know what this is. I just refactored it.

It was set to False already.

File: nrf_security/doc/prerequisites.rst
Comment (SebastianBoe) [reply nrf_security/doc/prerequisites.rst:4]: Changed to retain the docs on ARM_MBEDTLS_PATH.

**Comment by SebastianBoe** [comment]
@tejlmand : All feedback has been addressed. Please re-assess review.

File: nrf_security/README.rst
Comment (tejlmand) [inline nrf_security/README.rst:18]: Should we specify that the `ARM_MBEDTLS_PATH` should point to the top-level folder of the arm Mbed TLS repository ?
Or is that obvious.

@frkv thoughts ?

**Review by tejlmand — APPROVED** [review]
Thanks for putting this together :+1: 

A small comment to consider.

File: nrf_security/README.rst
Comment (SebastianBoe) [reply nrf_security/README.rst:18]: That is obvious IMHO.

**Comment by SebastianBoe** [comment]
@frkv : Can we have a +1 ?

File: nrf_security/CMakeLists.txt
Comment (frkv) [reply nrf_security/CMakeLists.txt:27]: Please Update comment

```
# MBEDTLS_PSA_CRYPTO_KEY_ID_ENCODES owner can't be set for Zephyr builds
# or when MBEDTLS_USE_PSA_CRYPTO is set (e.g. for TLS/DTLS and x.509 support)
# Note: This configuration is internal and may be removed with a new mbed TLS version
```

File: nrf_security/README.rst
Comment (frkv) [reply nrf_security/README.rst:18]: This should be obvious, agree with @SebastianBoe

**Review by frkv — APPROVED** [review]
LGTM

**Comment by SebastianBoe** [comment]
Merged without manifest PR to save time.

**Comment by greg-fer** [comment]
This PR slightly broke documentation structure. Kindly please do not merge doc-required PRs without an approval from a tech writer. Follow-up task: https://projecttools.nordicsemi.no/jira/browse/NCSDK-12851

**Comment by SebastianBoe** [comment]
@greg-fer : Noted. Sorry.

**Pull request merged**
